simpleviewinc/keg-herkin
# (Part 1) Zen 548 - side menu file tree

WIP
**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-548)

## Context

* We want to be able to load all the files within the mounted /tests folder. 
* update the side menu to load the mounted `tests` files


## Goal

* Update the API for an endpoint to load a file tree like data for the front-end to render
* setup front-end sidebar to show the content pulled from the api
* update settings modal to be dismissable (clicking the backdrop) if opening from settings icon
* PR pt2 will handle the interaction of loading the files onto the code editors

## Updates

* `package.json`
   * added `react-native-final-tree-view` module

* `repos/backend/endpoints/files.js`
  * add endpoint `get_tree` to return an array of objects  based on the file tree of mounted `/tests` folder

* `repos/backend/libs/fileSys/fileSys.js`
  * Migrated `keg-cli` file util here.
  * should be replaced by `keg-cli-utils` repo methods at some point

* `repos/backend/libs/fileSys/testFiles.js`
  * updated so you can pass in the full path or relative path

* `repos/tap/actions/*`   
  * added action to store the file tree
  * updated the `init` action to get the file tree data. also removed the example test data 
  * added action to be able to switch sidebar content. currently there's only 1 possible content: `file tree`

* `repos/tap/assets/icons/noteAdd.js`
   * note add icon

* `repos/tap/components/list/treeList/treeList.js`
  *  sets up the file tree component
  * it's a content type for side menu

* `repos/tap/components/modals/testSelectorModal.js`
  * updated so it can be dismissed on backdroptouch if it is not the initial modal

* `repos/tap/components/sidebar/sidebarContent.js`
  * manages the content displayed in the sidebar  

## Testing

### Setup
* Pull this branch
* Run `keg herkin && keg herkin start --config ./repos/example/herkin.config.js --from keg-herkin:zen-548-load-tests-tree`
  * *this will mount the example repo tests folder* 
* navigate to http://herkin-zen-548-load-tests-tree.local.kegdev.xyz/

### Test 1
* Dismiss the modal by Pressing `Do It` on the modal button (doesn't matter what options you pick)
* Toggle open the side menu
* Verify you see `bdd` and `waypoint` dropdown with the same files and folder structure as `repos/example/tests`
  * *empty folders and hidden files should not show up* 

### Test 2
* click on the `settings` icon (top right)
* verify that you can dismiss the modal by clicking the backdrop
